ui/client/components/graph/node-modal/editors/expression/SwitchIcon.js

```diff
@@ -1,18 +1,25 @@
 import React from "react"
 import PropTypes from "prop-types"
 import * as LoaderUtils from "../../../../../common/LoaderUtils"
+import {Types} from "./EditorType"
 
 export default function SwitchIcon(props) {
 
-  const {switchable, onClick, shouldShowSwitch, displayRawEditor, title, readOnly} = props
+  const {switchable, onClick, shouldShowSwitch, displayRawEditor, title, readOnly, fieldType} = props
 
-  function hint(switchable, displayRawEditor, readOnly) {
-    if (readOnly) {
+  function hint() {
+    if (readOnly)
       return "Switching to basic mode is disabled. You are in read-only mode"
-    } else {
-      return !displayRawEditor ? "Switch to expression mode" :
-        (switchable ? "Switch to basic mode" :
-          "Expression must be equal to true, false or be empty to switch to basic mode")
+
+    if (!displayRawEditor)
+      return "Switch to expression mode"
+
+    if (switchable) {
+      return "Switch to basic mode"
+    } else if (fieldType === Types.BOOLEAN || fieldType === Types.EXPRESSION) {
+      return "Expression must be equal to true or false to switch to basic mode"
+    } else if (fieldType === Types.STRING) {
+      return "Expression must be surrounded by single quotation marks or be empty to switch to basic mode"
```

Comment:
This is not true, "expression must be a simple string literal i.e. text surrounded by single quotation marks".

Reply:
Done